src/graphql/root/mutation/user-update-language.ts

```diff
@@ -15,24 +15,22 @@ const UserUpdateLanguageMutation = GT.Field({
   args: {
     input: { type: GT.NonNull(UserUpdateLanguageInput) },
   },
-  resolve: async (_, args, { user }) => {
+  resolve: async (_, args, { wallet }) => {
     const { language } = args.input
 
     if (language instanceof Error) {
       return { errors: [{ message: language.message }] }
     }
 
-    // TODO: redo this with app use-case
-    try {
-      user.language = language
-      await user.save()
-    } catch (err) {
-      return { errors: [{ message: err.message }] }
+    const result = wallet.updateLanguage({ language })
+
+    if (result instanceof Error) {
```

Comment:
if you dont call updateLanguage with await then result will be always a promise

Reply:
Fixed